src/osd/PG.cc

```diff
@@ -2662,10 +2665,13 @@ void PG::_update_calc_stats()
 	  misplaced += osd_objects;
       } else {
         // If this peer has more objects then it should, ignore them
-        int64_t osd_backfilled = MIN(num_objects, peer_info[p].stats.stats.sum.num_objects);
-        // Include computed backfilled objects on up nodes
-        object_copies += osd_backfilled;
-        backfilled += osd_backfilled;
+        int64_t osd_backfilled = MIN(num_objects,
+				     peer_info[p].stats.stats.sum.num_objects);
+	if (actingset.size() + num_backfill_shards_seen < pool.info.size) {
+	  // Include computed backfilled objects on up nodes
+	  object_copies += osd_backfilled;
+	}
+	backfilled += osd_backfilled;
```

Comment:
For the _misplaced_ calculation this should probably also be limited but based on number _acting_ and !_up_ OSDs.  It would no longer be the actual total backfilled.  Just because the first 10 objects of 3 backfilling OSDs have completed, doesn't mean that for 100 objects on 2 acting/!up (200 misplaced) should be reduced by 30 when those are the same 10 objects, it should be reduced by 20.  So misplaced will go from 200 to 0 only at the point that the backfill completes on at least 2 OSDs.